### PR TITLE
Cmake files improvements

### PR DESCRIPTION
**Pull request by adam-urbanczyk**
- [x] Removed custom find OCCT module
- [x] Bumped minimum cmake version to 3.17
- [x] Got  rid of `CONDA_PREFIX` references (see also #97)

### CODE REVIEW

**Comment by adam-urbanczyk** [comment]
@whophil

**Comment by whophil** [comment]
@adam-urbanczyk looks good visually! I am running the conda-forge builds using the new binding `CMakeLists.txt` file.
https://github.com/conda-forge/ocp-feedstock/pull/22

**Comment by whophil** [comment]
@adam-urbanczyk I'm not quite sure why, but using the updated CMakeLists.txt CMake looking for Boost during the build process when it didn't before. (see https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=642148&view=logs&j=d0d954b5-f111-5dc4-4d76-03b6c9d0cf7e&t=841356e0-85bb-57d8-dbbc-852e683d1642&l=390)

It isn't a problem in your CI because you build in an env with `boost=1.78` installed.

I wonder why this wasn't needed on the conda-forge build before, but is needed now.

**Comment by adam-urbanczyk** [comment]
AFAICT Boost is required by the generic `find_package( vtk )` call and is only a build time thing. I could not make the upstream `find_package( occt )` work with only specific vtk packages requested. It looks like the upstream-provided target definitions are not the best, but I'd prefer to not waste my time on maintaining custom target definitions. In the end if the price is one extra build-time dep, I don't see the problem.

See also: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=642148&view=logs&j=d0d954b5-f111-5dc4-4d76-03b6c9d0cf7e&t=841356e0-85bb-57d8-dbbc-852e683d1642&l=1033

File: templates/CMakeLists.j2 [hunk 1]
```diff
@@ -1,36 +1,29 @@
...
 find_package( pybind11 REQUIRED )
-find_package( OpenCascade REQUIRED COMPONENTS OPENCASCADE )
-find_package( VTK 9 REQUIRED COMPONENTS WrappingPythonCore RenderingCore CommonDataModel CommonExecutionModel )
+find_package( VTK REQUIRED )
+find_package( OpenCASCADE REQUIRED )
 
 include_directories( ${PROJECT_SOURCE_DIR} )
 file( GLOB CPP_FILES ${PROJECT_SOURCE_DIR}/*.cpp )
 
 add_library( {{ name }} MODULE ${CPP_FILES} )
```
Comment (whophil) [inline templates/CMakeLists.j2:23]: Have you tried `pybind11_add_module` in place of this?

**Review by whophil — APPROVED** [review]

**Comment by whophil** [comment]
> AFAICT Boost is required by the generic `find_package( vtk )` call and is only a build time thing. I could not make the upstream `find_package( occt )` work with only specific vtk packages requested... In the end if the price is one extra build-time dep, I don't see the problem.

Agreed. Added boost 1.78 to the build deps (and one other change - see comments) and the conda-forge build seems to be running OK.

File: templates/CMakeLists.j2 (hunk 1 above)
Comment (adam-urbanczyk) [reply templates/CMakeLists.j2:23]: Yes, I prefer to have explicit control on target creation.

File: templates/CMakeLists.j2 [hunk 2]
```diff
@@ -1,36 +1,29 @@
...
 
-target_link_libraries( {{ name }} PRIVATE OPENCASCADE pybind11::pybind11 VTK::WrappingPythonCore VTK::RenderingCore VTK::CommonDataModel VTK::CommonExecutionModel)
+target_link_libraries( {{ name }} PRIVATE ${OpenCASCADE_LIBRARIES} )
+target_link_libraries( {{ name }} PRIVATE pybind11::pybind11 VTK::WrappingPythonCore VTK::RenderingCore VTK::CommonDataModel VTK::CommonExecutionModel)
```
[the remark this answers is not in the document]
Comment (adam-urbanczyk) [reply templates/CMakeLists.j2:26]: Good catch

**Pull request merged**